rakam-spi/src/main/java/org/rakam/analysis/AbstractFunnelQueryExecutor.java

```diff
@@ -156,4 +177,9 @@ protected boolean testDeviceIdExists(FunnelStep firstAction, Map<String, List<Sc
         }
         return schemaFields.stream().anyMatch(e -> e.getName().equals("_device_id"));
     }
+
+    public void setTimeStampMapping(Map<FunnelTimestampSegments, String> timeStampMapping) {
+        this.timeStampMapping = timeStampMapping;
+    }
```

Comment:
You don't need to have this method, you can mark `timeStampMapping ` as `protected`.